### ISSUE TITLE
Provide better experience for NickServ 

### ISSUE THREAD

**Issue by csoriano89** [matrix-org/matrix-appservice-irc#458]
Original bug report at https://github.com/vector-im/riot-web/issues/4356

Description

Currently when a room is register only (+r) an error is displayed
![matrix registered nick](https://user-images.githubusercontent.com/1755514/27283788-3efa6442-54f6-11e7-8253-df9637a15ac8.png)
However, there are no instructions on what to do in that situation, which is really hard for a new user to figure out (yay IRC).
Would it be possible to add a link that opens a chat to nickserv with simple instructions, or a link to https://github.com/matrix-org/matrix-appservice-irc/wiki/End-user-FAQ#how-do-i-registeridentify-to-nickserv so the user can figure it out?

Steps to reproduce

Join any register only channel, like #fedora in freenode.

**Comment by Mikaela** [comment]
I think more descriptive issue title might be *Handle ERR_NEEDREGGEDNICK and give the user insturctions on what to do*.

**Comment by Kegsay** [comment]
We should publish the set of error codes which the IRC bridge can return as the `reason` field for kicks. However, making it clearer is not the bridge's responsibility. Riot will need to check for the error code and then display i18n'd help text.

Renaming issue accordingly.

**Comment by syam3526** [comment]
please provide instructions on how to link nickserv with matrix or how  to use nickerv as the identity server instead of vector

**Comment by ssbarnea** [comment]
true, I was hit recently by this and it is total mistery. I lost more than 3 hours trying to get back in and still counting as even after doing "!storepass .." it does not work and matrix does not reply with any useful information on how to re-authenticate.
```
fn_bot: "Successfully stored password on chat.freenode.net. When you next reconnect to chat.freenode.net, this password will be automatically sent in a PASS command which most IRC networks will use as your NickServ password. This means you will not need to talk to NickServ. This does NOT apply to your currently active connection: you still need to talk to NickServ one last time to authenticate your current connection if you haven't already."
```

**Comment by ssbarnea** [comment]
... And the nightmare started again yesterday. I was kicked out from lots of freenode rooms without notification or solution for rejoin them. After one day I am still out and using a normal IRC client as I wasn't able to make matrix/riot to work.

**Comment by cuibonobo** [comment]
I was having trouble authenticating with appservice-irc and Nickserv because it seemed like the bots weren't reading my messages. I ultimately fixed this by leaving both rooms (and even 'forgetting' them from the historical rooms list) and starting the process from scratch.

**Comment by JaneSmith** [comment]
I'd also like to see this improved. I tried to join the Fedora room just like the OP. I got kicked and told I need a registered nick. I read through the wiki page link. And I still have no clue what I'm supposed to do. I'm just here scratching my head wondering what to do. These instructions are not clear.

**Comment by PureBasic** [comment]
I just started using Riot today as a replacement for Pidgin. NickServ is not responding to any commands...
I don't know what is my status, IRC nick, or anything, and cannot query...
Back to Pidgin :(
